src/mappings/Competition/mapping.ts

```diff
@@ -100,6 +140,31 @@ export function handleNewVote(event: NewVote): void {
       suggestion.save();
     }
     vote.save();
+
+    let competition = Competition.bind(event.address);
+    competitionProposal.winningSuggestions = [];
+    let winningSuggestions = competitionProposal.winningSuggestions;
+    debug(competitionProposal.suggestions.toString() + '');
+    for (let i = 0; i < competitionProposal.suggestions.length; i++) {
+      let suggestions = competitionProposal.suggestions;
+      if (suggestions != null) {
+        let currentSuggestionId = (suggestions as string[])[i];
+        let competitionSuggestion = CompetitionSuggestion.load(currentSuggestionId as string);
+        if (competitionSuggestion != null) {
+          let index = competition.getOrderedIndexOfSuggestion(competitionSuggestion.suggestionId);
+          if (index >= competitionProposal.numberOfWinners ||
+            competitionSuggestion.totalVotes.equals(BigInt.fromI32(0))) {
+            competitionSuggestion.positionInWinnerList = null;
+          } else {
+            competitionSuggestion.positionInWinnerList = index;
+            winningSuggestions.push(currentSuggestionId as string);
+          }
+          competitionSuggestion.save();
+        }
+      }
+    }
+    competitionProposal.winningSuggestions = winningSuggestions;
+    competitionProposal.save();
```

Comment:
if the winningSuggestions is not updated due to the vote so there is no need to re save.

Reply:
This will just complexity to save the array and compare.